src/deepsparse/v2/pipeline.py

```diff
@@ -61,85 +62,66 @@ def __init__(
 
         self._scheduler_group = SchedulerGroup(self.schedulers)
 
-    def _run_sequential(
+    async def run_async(
         self,
-        inp: Any,
+        *args,
         inference_state: InferenceState,
         pipeline_state: PipelineState,
-        start: str,
-        end: str,
+        **kwargs,
     ):
-        next_step = start
-        while next_step != end:
-            outputs = self._run_next_step(
-                func=self.ops[next_step],
-                next_step=next_step,
-                input=inp,
-                pipeline_state=pipeline_state,
-                inference_state=inference_state,
-            )
-            next_step, operator_output, state_update = outputs
-            if state_update:
-                inference_state.update_state(state_update)
-            inp = operator_output
-        return inp
-
-    def _apply_split(self, inp: Any, inference_state: InferenceState):
         """
-        Split inputs using the pipeline's expand_inputs function. Inputs are split
-        into a batch size of one when a SPLIT_ROUTE node is found in a given pipeline's
-        provided router. The split batches are run asynchronously and then joined when
-        a JOIN_ROUTE node is found, using the pipeline's condense_inputs function.
+        Run through the operators using the provided router and scheduler.
+        The input to a given operator is the output of the previous operator.
+
+        :param inference_state: inference_state for the pipeline.
+        :param pipeline_state: pipeline_state for the pipeline. The values in the state
+            are created during pipeline creation and are read-only during inference.
         """
+        loop = asyncio.get_event_loop()
```

Comment:
I meant to ask you at some point, but forgot... Is there a reason to use the event loop that is registered with `asyncio`? You could create a `deepsparse` specific event loop. That said, I don't know what the tradeoffs of that choice would be.

Here's an example of creating an independent event loop:
```python
# Share one thread pool across all instances. Lifetime is equal to application lifetime.
_thread_pool_executor: ThreadPoolExecutor = ThreadPoolExecutor(max_workers=4)
# Use a new event loop to avoid saturating the default event loop
_loop = asyncio.new_event_loop()
_loop.set_default_executor(_thread_pool_executor)
```

While trying to answer the question of when you should create a new event loop, I found that the docs suggest using `asyncio.get_running_loop` instead of `asyncio.get_event_loop` when calling from a co-routine or callback:

> Because this function has rather complex behavior (especially when custom event loop policies are in use), using the [get_running_loop()](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_running_loop) function is preferred to [get_event_loop()](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop) in coroutines and callbacks.
>
> https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop